### DISCUSSION TITLE
Custom response breaks for Starlite >= 1.48.0

### DISCUSSION THREAD

**Discussion by david-welch** [litestar-org/litestar#1072] [Q&A] [answered]
We have a custom response class to handle serialization of Mongo IDs: 
```python
class BeanieResponse(Response):
    def serializer(self, value: Any) -> Dict[str, Any]:
        if isinstance(value, PydanticObjectId):
            return str(value)
        if isinstance(value, Link):
            return value.to_dict().get("id")
        if isinstance(value, Int64):
            return int(value)
        return super().serializer(value)
```
On upgrading Starlite to 1.48.0 or greater, I get the following error: 
```
  File "<PATH>/venv/lib/python3.10/site-packages/starlite/response/base.py", line 286, in render
    return encode_json(content, self._enc_hook)
  File "<PATH>/venv/lib/python3.10/site-packages/starlite/utils/serialization.py", line 108, in encode_json
    return msgspec.json.encode(obj, enc_hook=enc_hook)
TypeError: BeanieResponse.serializer() missing 1 required positional argument: 'value'
```

I see this info in the method `Set `type_encoders` on a higher layer instead or pass `type_encoders` to the `Response` instance directly`. I'm unclear how to make use of attribute or if this is the way to solve the problem. Any ideas?

**Comment by provinzkraut** [comment, marked as answer]
`Response.serializer` is a classmethod, so you need to add the `@classmethod` decorator.

However, there are easier methods to solve this! Take a look at the [custom responses](https://starlite-api.github.io/starlite/latest/usage/5-responses/10-custom-responses/) examples in the docs. You can add a `type_encoders` dictionary on any layer of the application, so a minimal implementation would look something like this:

```python
from starlite import Starlite
# add your type imports here

beanie_encoders = {
    PydanticObjectId: str,
    Link: lambda link: link.to_dict().get("id"),
    Int64: int,
}

app = Starlite(..., type_encoders=beanie_encoders)
```

**Reply by david-welch** [reply]
Thanks! This solves it.
